crates/router/src/analytics.rs

```diff
@@ -1864,6 +1864,7 @@ pub mod routes {
                     .global_store
                     .list_user_roles_by_user_id(ListUserRolesByUserIdPayload {
                         user_id: &auth.user_id,
+                        tenant_id: auth.tenant_id.as_ref().unwrap_or(&state.tenant.tenant_id),
```

Comment:
If default value if taken for tenant id not being present in auth token, then won't a user be able to access any tenancy if user gets access to a token with `tenant_id: None`.
Are we ensuring tenant_id will never be null ?